src/hotspot/share/classfile/loaderConstraints.cpp

```diff
@@ -62,30 +62,35 @@ class LoaderConstraint : public CHeapObj<mtClass> {
 
   int num_loaders() const { return _loaders->length(); }
   ClassLoaderData* loader_data(int i) { return _loaders->at(i); }
-  void set_loader_data(ClassLoaderData* p) { _loaders->push(p); }
+  void add_loader_data(ClassLoaderData* p) { _loaders->push(p); }
 
   void remove_loader_at(int n) {
     assert(_loaders->at(n)->is_unloading(), "should be unloading");
     _loaders->remove_at(n);
   }
 
   // convenience
-  void set_loader(oop p) {
+  void add_loader(oop p) {
     _loaders->push(ClassLoaderData::class_loader_data(p));
   }
 };
 
-class LoaderConstraintEntry : public StackObj {      // copied into hashtable as value
+// For this class name, these are the LoaderConstraints for classes loaded with this name.
+class ClassNameEntry {                               // copied into hashtable as value
```

Comment:
I find the naming for these map/table entries very awkward in general. A Map is a mapping from keytype K to a value type V. Here K ==  Symbol* (a class name), and V == a set of LoaderConstraints

Reply:
I was trying to not repeat "LoaderConstraint" in the name of the hashtable entry of LoaderConstraints, because that's what made it look confusing (especially in the diff).
SetOfLoaderConstraints or LoaderConstraintSet.  Maybe ClassNameEntry is a terrible name.  I'm open for suggestions.

Reply:
How about ConstraintSet ?